init.ts

```diff
@@ -0,0 +1,231 @@
+        example: 1,
+      },
+      currentPageSize: {
+        type: 'integer',
+        description: 'Number of results per page',
+        example: 25,
+      },
+    },
+  },
+  PaginationLinks: {
+    properties: {
+      first: {
+        type: 'string',
+        format: 'uri',
+        description: 'The first page of data',
+      },
+      last: {
+        type: 'string',
+        format: 'uri',
+        description: 'The last page of data',
+      },
+      prev: {
+        type: 'string',
+        format: 'uri',
+        description: 'The previous page of data',
+      },
+      next: {
+        type: 'string',
+        format: 'uri',
+        description: 'The next page of data',
+      },
+    },
+  },
+};
+
+const paginationParameters = {
+  pageNumber: {
+    name: 'page[number]',
+    in: 'query',
+    required: false,
+    description: 'Page number of results',
+    schema: {
+      type: 'integer',
+      minimum: 1,
+      default: 1,
+    },
+  },
+  pageSize: {
+    name: 'page[size]',
+    in: 'query',
+    required: false,
+    description: 'Number of results to return',
+    schema: {
+      type: 'integer',
+      minimum: 1,
+      maximum: 500,
+      default: 25,
+    },
+  },
+};
+
+const errorSchema = {
+  content: {
+    'application/json': {
+      schema: {
+        $ref: '#/components/schemas/ErrorResult',
+      },
+    },
+  },
+};
+
+/**
+ * Creates the initial openapi object template
+ *
+ * @param config - The config file object
+ * @returns The updated openapi object
+ */
+const init = (config: any) => {
+  const containsPaginated = _.some(config.resources, (resource) => resource.paginate);
+  const extraSchemas = containsPaginated ? paginationSchemas : {};
+  const extraParameters = containsPaginated ? paginationParameters : {};
+
+  const openapi = {
+    openapi: '3.0.0',
+    info: {
+      title: config.title,
+      description: 'REPLACEME',
+      version: config.version,
+      license: {
+        name: 'REPLACEME',
+        url: 'REPLACEME',
+      },
+      contact: {
+        name: 'IS Data Architecture Team',
+        url: 'https://is.oregonstate.edu/data-architecture',
+        email: 'isdataarchitecture@oregonstate.edu',
+      },
+    },
+    externalDocs: {
+      description: 'GitHub Repository',
+      url: config.githubUrl,
+    },
+    servers: [
+      {
+        url: `https://api.oregonstate.edu/${config.version}`,
+      },
+    ],
+    security: [
+      {
+        OAuth2: ['full'],
+      },
+    ],
+    paths: {},
+    components: {
+      securitySchemes: {
+        OAuth2: {
+          type: 'oauth2',
+          flows: {
+            clientCredentials: {
+              tokenUrl: 'https://api.oregonstate.edu/oauth2/token',
+              scopes: {
+                full: 'Full access to the API',
+              },
+            },
+          },
+        },
+      },
+      parameters: extraParameters,
+      responses: {
+        400: {
```

Comment:
These keys are strings. You can do something like this in a node session:
```
> { 400: 'val' }
{ '400': 'val' }
```

Also from [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Objects_and_properties):
> Please note that all keys in the square bracket notation are converted to string unless they're Symbols, since JavaScript object property names (keys) can only be strings or Symbols (at some point, private names will also be added as the class fields proposal progresses, but you won't use them with [] form). For example, in the above code, when the key obj is added to the myObj, JavaScript will call the obj.toString() method, and use this result string as the new key.

Reply:
Oh yeah you're right, sometimes it's really easy to get confused when you try to switch between javascript and other languages. Feel free to ignore my comment here.

Reply:
Right, I wasn't sure about this initially either since I get different syntax highlighting for keys that appear to be numbers, haha.